src/Angular.js

```diff
@@ -1517,7 +1517,8 @@ function allowAutoBootstrap(document) {
     var link = document.createElement('a');
     link.href = src.value;
 
-    if (document.location.origin === link.origin) {
+    if (document.location.protocol === link.protocol && document.location.host === link.host &&
```

Comment:
```
if (document.location.protocol === link.protocol && document.location.host === link.host &&
      document.location.port === link.port) {
      // Same-origin resources are always allowed, even for non-whitelisted schemes.
      return true;
    }
```

Doesn't checking the `host` already include comparing the `port`? Maybe you where looking for `hostname` (https://developer.mozilla.org/en-US/docs/Web/API/Location)?

What about keep checking `origin`, but provide a fallback (polyfill?) when origin doesn't exist ?
```
if (!window.location.origin) {
  window.location.origin = window.location.protocol + "//" + window.location.hostname + (window.location.port ? ':' + window.location.port: '');
}
```